testbed/testbed/data_providers.go

```diff
@@ -130,10 +130,18 @@ func (dp *perfTestDataProvider) GenerateMetrics() (pdata.Metrics, bool) {
 			dataPoint.SetStartTimestamp(pdata.TimestampFromTime(time.Now()))
 			value := dp.dataItemsGenerated.Inc()
 			dataPoint.SetValue(int64(value))
-			dataPoint.LabelsMap().InitFromMap(map[string]string{
-				"item_index":  "item_" + strconv.Itoa(j),
-				"batch_index": "batch_" + strconv.Itoa(int(batchIndex)),
-			})
+			var labelsMap map[string]string
+			if dp.options.IsScraping {
+				labelsMap = map[string]string{
+					"item_index": "item_" + strconv.Itoa(j),
+				}
+			} else {
+				labelsMap = map[string]string{
+					"item_index":  "item_" + strconv.Itoa(j),
+					"batch_index": "batch_" + strconv.Itoa(int(batchIndex)),
+				}
+			}
```

Comment:
Why is this change necessary?

Reply:
Including the batch index means adding new dimensions for the metric each time the data is sent. The Prometheus /metrics endpoint that's scraped will keep all timeseries, so the amount of data doubles each time new data is sent instead of keeping the same dimensions and a set amount of data to scrape. I have added a comment to explain this.